src/hotspot/cpu/x86/stubGenerator_x86_64.cpp

```diff
@@ -5360,6 +5360,378 @@ address generate_avx_ghash_processBlocks() {
     return start;
   }
 
+  /* The 2-block AVX/AVX2-enabled ChaCha20 block function implementation */
+  address generate_chacha20Block_avx() {
```

Comment:
Considering you already introduce a dedicated CPP file, it makes sense to move the guts of this function into `macroAssembler_x86_chapoly.cpp`.

Reply:
I've updated the code to follow your example with AES and moved the intrinsics into their own stubGenerator_x86_64_chapoly.cpp.  I hope that will compartmentalize things.  I'm not sure if I should combine the macroAssembler_x86_64_chapoly.cpp with the new stubGenerator file.  Certainly willing to do that, but I wanted to get the dedicated stubGenerator file working first since the last merge was ugly.  Going forward merges should be much easier now that my code is compartmentalized.